test/gtest/pad_constant.hpp

```diff
@@ -0,0 +1,218 @@
+    {
+        return os << "(N: " << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H
+                  << " W:" << tc.W << ")";
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        if((N != 0) && (C != 0) && (D != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, D, H, W});
+        }
+        else if((N != 0) && (C != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, 1, H, W});
+        }
+        else if((N != 0) && (C != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, 1, 1, W});
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, 1, 1, 1, W});
+        }
+        else if(N != 0)
+        {
+            return std::vector<size_t>({N, 1, 1, 1, 1});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+};
+
+std::vector<PadConstantTestCase> PadConstantTestConfigs()
+{
+    return {{8, 120, 1, 1, 1},
+            {8, 120, 1, 1, 1},
+            {8, 1023, 1, 1, 1},
+            {8, 1024, 1, 1, 768},
+            {8, 1023, 1, 1, 1},
+            {8, 1024, 1, 1, 768},
+            {16, 1024, 1, 1, 768},
+            {16, 1024, 1, 1, 768},
+            {48, 8, 1, 512, 512},
+            {48, 8, 1, 512, 512},
+            {16, 311, 1, 98, 512},
+            {16, 311, 1, 98, 512}};
+}
+
+inline std::vector<size_t> GetStrides(std::vector<size_t> input, bool contiguous)
+{
+    if(!contiguous)
+        std::swap(input.front(), input.back());
+    std::vector<size_t> strides(input.size());
+    strides.back() = 1;
+    for(int i = input.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * input[i + 1];
+    if(!contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename T>
+struct PadConstantTest : public ::testing::TestWithParam<PadConstantTestCase>
+{
+protected:
+    PadConstantTestCase pad_constant_config;
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> backward_output;
+
+    tensor<T> ref_output;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+    miopen::Allocator::ManageDataPtr backward_output_dev;
+
+    size_t padding[10];
+
+    void SetUp() override
+    {
+        auto&& handle       = get_handle();
+        pad_constant_config = GetParam();
+        auto gen_value      = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+
+        auto in_dims = pad_constant_config.GetInput();
+
+        auto strides = GetStrides(in_dims, false);
+        input        = tensor<T>{in_dims, strides}.generate(gen_value);
+
+        input_dev = handle.Write(input.data);
+
+        memset(padding, 0, sizeof(padding));
+        // Generate random padding for the first 3 dims
```

Comment:
Is there a reason to only insert random values from 2 to 8?

Reply:
It's because of `IsImprovementOverROCm`.